### PR TITLE
Include the configuration condition for ExternalWarningLevel.

### PR DESCRIPTION
**Pull request by LoneBoco**
**What does this PR do?**

Fixes a bug with ExternalWarningLevel where the entry would be duplicated for every configuration you had.

Before:
```xml
<ClCompile Include="file.cpp">
  <ExternalWarningLevel>Level3</ExternalWarningLevel>
  <ExternalWarningLevel>Level3</ExternalWarningLevel>
  <ExternalWarningLevel>Level3</ExternalWarningLevel>
  <ExternalWarningLevel>Level3</ExternalWarningLevel>
</ClCompile>
```   
After:
```xml
<ClCompile Include="file.cpp">
  <ExternalWarningLevel>Level3</ExternalWarningLevel>
</ClCompile>
```

**How does this PR change Premake's behavior?**

No breaking changes.

**Anything else we should know?**

The bug was introduced with changeset 065b3acb2fdbb137e95fa1d59f4f37c5e4b334e6.

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [x] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: modules/vstudio/vs2010_vcxproj.lua [hunk 1]
```diff
@@ -2854,10 +2854,10 @@
...
 
 
-	function m.externalWarningLevel(cfg)
+	function m.externalWarningLevel(cfg, condition)
 		if _ACTION >= "vs2022" then
 			local map = { Off = "TurnOffAllWarnings", High = "Level4", Extra = "Level4", Everything = "Level4" }
-			m.element("ExternalWarningLevel", nil, map[cfg.externalwarnings] or "Level3")
+			m.element("ExternalWarningLevel", condition, map[cfg.externalwarnings] or "Level3")
 		end
 	end
```
Comment (Jarod42) [inline modules/vstudio/vs2010_vcxproj.lua:2863]: Below `function m.externalAngleBrackets(cfg)` is also missing `condition` parameter.

**Review by Jarod42 — COMMENTED** [review]
You checked *"Add Unit tests"*, but don't provide tests to show old bug.

**Comment by LoneBoco** [comment]
Okay, I fixed all instances of file level handling of `externalwarnings` and `externalanglebrackets`, including adding tests for them.

**Comment by LoneBoco** [comment]
I also prevented it from exporting on the file level if it wasn't declared.

File: modules/vstudio/vs2010_vcxproj.lua [hunk 2]
```diff
@@ -2862,7 +2862,28 @@
...
+
+
 	function m.externalAngleBrackets(cfg)
+		if _ACTION >= "vs2022" then
+			if cfg.externalanglebrackets == p.OFF then
+				m.element("TreatAngleIncludeAsExternal", nil, "false")
+			elseif cfg.externalanglebrackets == p.ON then
+				m.element("TreatAngleIncludeAsExternal", nil, "true")
+			end
+		end
```
Comment (Jarod42) [inline modules/vstudio/vs2010_vcxproj.lua:2882]: Possibly just call `externalAngleBracketsFile(cfg, nil)`?

Comment (LoneBoco) [reply modules/vstudio/vs2010_vcxproj.lua:2882]: Done.

**Review by nickclark2016 — APPROVED** [review]

**Pull request merged**